File: lc807.py
class Solution(object):
    def maxIncreaseKeepingSkyline(self, grid):
        """
        :type grid: List[List[int]]
        :rtype: int
        """
        bottom = []
        left = []
        output = 0
        for i in range(len(grid)):
            left.append(max(grid[i]))
        for i in range(len(grid[0])):
            tmp = []
            for j in range(len(grid)):
                tmp.append(grid[j][i])
            bottom.append(max(tmp))
        #print bottom
        #print left
        
        for i in range(len(grid)):
            for j in range(len(grid[0])):
                if grid[i][j] < left[i] and grid[i][j] < bottom[j]:
                    tmp_v = left[i] if left[i] < bottom[j] else bottom[j]
                    output += (tmp_v - grid[i][j])
        return output

File: lc807.c
int maxIncreaseKeepingSkyline(int** grid, int gridRowSize, int *gridColSizes) {
    printf("%d", *gridColSizes);
    int max[gridRowSize][*gridColSizes];
    int max_r[gridRowSize];
    int max_c[*gridColSizes];
    int tmp = 0, sum = 0;
    for(int i = 0; i < gridRowSize; i++)
    {
        tmp = 0;
        for(int j = 0; j < *gridColSizes; j++)
        {
            if(tmp < grid[i][j])
                tmp = grid[i][j];
        }
        max_r[i] = tmp;
    }
    for(int i = 0; i < *gridColSizes; i++)
    {
        tmp = 0;
        for(int j = 0; j < gridRowSize; j++)
        {
            if(tmp < grid[j][i])
                tmp = grid[j][i];
        }
        max_c[i] = tmp;
    }
    for(int i = 0; i < gridRowSize; i++)
    {
        for(int j = 0; j < *gridColSizes; j++)
        {
            if(max_r[i] > max_c[j])
                sum += max_c[j] - grid[i][j];
            else
                sum += max_r[i] - grid[i][j];
            
        }
    }
    return sum;
}